test/e2e/customconfigs/waitforpodsready_test.go

```diff
@@ -0,0 +1,328 @@
+				job = testingjob.MakeJob("job-timeout", ns.Name).
+					Queue(lq.Name).
+					Request(corev1.ResourceCPU, "2").
+					Parallelism(1).
+					Obj()
+				gomega.Expect(k8sClient.Create(ctx, job)).Should(gomega.Succeed())
+			})
+
+			wlKey = types.NamespacedName{
+				Name:      workloadjob.GetWorkloadNameForJob(job.Name, job.UID),
+				Namespace: ns.Name,
+			}
+
+			ginkgo.By("waiting for the workload to be created and verifying it is not admitted", func() {
+				var wl kueue.Workload
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, &wl)).Should(gomega.Succeed())
+
+					g.Expect(wl.Status.Admission).To(gomega.BeNil())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("waiting for the workload to be evicted", func() {
+				var wl kueue.Workload
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, &wl)).Should(gomega.Succeed())
+
+					podsReadyCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadPodsReady)
+					g.Expect(podsReadyCond).ShouldNot(gomega.BeNil())
+					g.Expect(podsReadyCond.Status).To(gomega.Equal(metav1.ConditionFalse))
+					g.Expect(podsReadyCond.Reason).To(gomega.Equal(kueue.WorkloadWaitForStart))
+
+					evictedCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadEvicted)
+					g.Expect(evictedCond).ShouldNot(gomega.BeNil())
+					g.Expect(evictedCond.Status).To(gomega.Equal(metav1.ConditionTrue))
+					g.Expect(evictedCond.Reason).To(gomega.Equal(kueue.WorkloadEvictedByPodsReadyTimeout))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("verifying that the job is suspended", func() {
+				var updatedJob batchv1.Job
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(job), &updatedJob)).Should(gomega.Succeed())
+					g.Expect(ptr.Deref(updatedJob.Spec.Suspend, false)).To(gomega.BeTrue())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("verifying that the workload is requeued", func() {
+				var wl kueue.Workload
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, &wl)).Should(gomega.Succeed())
+					g.Expect(wl.Status.RequeueState).ShouldNot(gomega.BeNil())
+					g.Expect(*wl.Status.RequeueState.Count).To(gomega.Equal(int32(1)))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("simulating a second eviction cycle to check workload deactivation", func() {
+				var wl kueue.Workload
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, &wl)).Should(gomega.Succeed())
+					g.Expect(wl.Status.RequeueState).Should(gomega.BeNil())
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+		})
+	})
+
+	ginkgo.When("WaitForPodsReady has default Timeout and a short RecoveryTimeout", func() {
```

Comment:
Is there a reason we use short recoveryTimeout instead of a tiny one?

Reply:
Have feared that it would fail, however after checking seems fine, so tiny timeout it is